docs/GHA.md

```diff
@@ -0,0 +1,50 @@
+
+The `cancel-in-progress` Boolean value that can be defined in a `concurrency` hash inside a GitHub Action job workflow maps to the Buildkite pipeline setting of [Cancel Intermediate Builds](https://buildkite.com/docs/pipelines/skipping#cancel-running-intermediate-builds) - which can be set within a pipeline's settings page, or when creating/updating a pipeline via the [REST](https://buildkite.com/docs/apis/rest-api/pipelines#create-a-yaml-pipeline)/[GraphQL](https://buildkite.com/docs/apis/graphql/schemas/mutation/pipelinecreate) APIs.
+
+### env 
+
+GitHub Actions allows the specification of environment variables at the workflow level, at a `job` level, and for each `step` within a `job`.
+
+Environment variables that are defined at the top of a workflow will be transition to [build level](https://buildkite.com/docs/pipelines/environment-variables#environment-variable-precedence) environment variables in the generated Buildkite pipeline. Environment variables defined within the context of each of a workflows' `jobs` are transitioned to [step level](https://buildkite.com/docs/pipelines/environment-variables#runtime-variable-interpolation) environment variables.
+
+### jobs
+
+GitHub Action workflows allows you to specify one or more `jobs` - main tasks of a workflow that run in parallel by default.
+
+The Buildkite Migration tool currently supports the following 
+
+- `runs_on`: The `runs_on` key defines the type of machine that the job will run on. Within Buildkite, this is mapped to an agent targeting [tag](https://buildkite.com/docs/agent/v3/queues#targeting-a-queue) of `runs_on`. Note that jobs that target custom `tag` names will have a `queue` target of `default`.
+- `steps`: Steps that are defined for a particular `job`. Any `run` key is supported - and at present, any action that defines a `uses` attribute currently is not supported.
+
+## Unsupported properties
+
+### defaults
+
+A `defaults` key allows the definition of a hash in which configuration will apply to all of a workflow's jobs. 
+
+While there isn't a direct transition from a GitHub Action workflow to a Buildkite pipeline - the Buildkite platform allows jobs to have default configuration applied to them through the use of [Agent](https://buildkite.com/docs/agent/v3/hooks#agent-lifecycle-hooks) or [Job](https://buildkite.com/docs/agent/v3/hooks#job-lifecycle-hooks) Lifecycle hooks. Agent lifecycle hooks allows the configuration of customized commands on agent start up and shut down. Job Lifecycle hooks allow for the customization of what occurs at each stage of a job's lifecycle - and can be specified and applied at an agent, repository or plugin level.
```

Comment:
I'm not sure that this should be on the _unsupported_ list as there is a [partial (and untested) implementation of `defaults` ](https://github.com/buildkite/migration/blob/main/app/lib/bk/compat/parsers/gha.rb#L41-L43)

That said, I am not sure why you are talking about hooks (unless it is to set environment variables.

Reply:
I'm not sure that a default is the same as a hook? It feels more like YAML anchors than hooks?

Reply:
Adjusted with both - I've added YAML anchors as the primary alternative - hooks below

Reply:
This also is now partially supported (instead of being under unsupported)